src/cargo/core/resolver/conflict_cache.rs

```diff
@@ -1,40 +1,98 @@
-use std::collections::{HashMap, HashSet};
+use std::collections::{BTreeMap, HashMap, HashSet};
 
 use super::types::ConflictReason;
 use core::resolver::Context;
 use core::{Dependency, PackageId};
 
+enum ConflictStore {
+    Con(BTreeMap<PackageId, ConflictReason>),
+    Map(HashMap<PackageId, ConflictStore>),
+}
+
+impl ConflictStore {
+    /// Finds any known set of conflicts, if any,
+    /// which are activated in `cx` and pass the `filter` specified?
+    pub fn find_conflicting<F>(
+        &self,
+        cx: &Context,
+        filter: &F,
+    ) -> Option<&BTreeMap<PackageId, ConflictReason>>
+    where
+        for<'r> F: Fn(&'r &BTreeMap<PackageId, ConflictReason>) -> bool,
+    {
+        match self {
+            ConflictStore::Con(c) => {
+                if filter(&c) {
+                    Some(c)
+                } else {
+                    None
+                }
+            }
+            ConflictStore::Map(m) => {
+                for (pid, store) in m {
+                    if cx.is_active(pid) {
```

Comment:
I'm trying to piece together why this is here a bit. This call to `is_active` wasn't present in the previous version. In the old versiom there's also no call to `is_conflicting` any more, right?

I feel like this is correct though because if `is_active` returns false then there's no need to look at any of the conflict caches in the subtree because they're guaranteed to not fit our current context as one of their candidates isn't active.

Can you elaborate more though on what happened to the call to `is_conflicting`?